prime-angular-frontend/src/app/modules/site-registration/shared/components/summary-card/summary-card.component.ts

```diff
@@ -1,12 +1,35 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnInit, Output, EventEmitter, Input, TemplateRef } from '@angular/core';
 
 @Component({
   selector: 'app-summary-card',
   templateUrl: './summary-card.component.html',
   styleUrls: ['./summary-card.component.scss']
 })
 export class SummaryCardComponent implements OnInit {
-  constructor() { }
+  @Input() public icon: string;
+  @Input() public title: string;
+  @Input() public menu: TemplateRef<any>;
+  @Input() public menuOutletContext: { [key: string]: any } | null;
+  @Input() public properties: [string, string];
+  @Input() public actionButtonTitle: string;
+  @Input() public actionDisabled: boolean;
+
+  @Output() public action: EventEmitter<void>;
+  @Output() public remove: EventEmitter<void>;
+
+  constructor() {
+    this.action = new EventEmitter<void>();
+    this.remove = new EventEmitter<void>();
+  }
+
+  public onClick() {
```

Comment:
`onClick` doesn't really provide context other than you clicked something in the DOM, but the element it references uses `actionButtonTitle`, `actionDisabled`, and the event handler itself invokes the `action` event emitter.  I'd probably call this event handler `onAction`